autonomy/deploy/base.py

```diff
@@ -126,12 +134,18 @@ def generate_common_vars(self, agent_n: int) -> Dict:
             agent_vars["AEA_PASSWORD"] = self.private_keys_password
         return agent_vars
 
-    def generate_agent(self, agent_n: int) -> Dict[Any, Any]:
+    def generate_agent(
+        self, agent_n: int, override_idx: Optional[int] = None
+    ) -> Dict[Any, Any]:
         """Generate next agent."""
         agent_vars = self.generate_common_vars(agent_n)
         if len(self.service.overrides) == 0:
             return agent_vars
-        overrides = self.process_model_args_overrides(agent_n)
+
+        if override_idx is None:
+            override_idx = agent_n
+
+        overrides = self.process_model_args_overrides(override_idx)
```

Comment:
We define override index using the agent instances list and export them using respective agent index.